### ISSUE TITLE
Totals for a cash transaction offset (CCs/CIs) get cleared when opened on server

### ISSUE THREAD

**Issue by DhanyaHerath** [openmsupply/mobile#2242]
## Describe the bug
Wasn't sure whether this should be on desktop issue or mobile, but this doesn't happen for transactions made on desktop: 

When I make a transaction on the cash register in mobile, I can see it on desktop: 
![image](https://user-images.githubusercontent.com/29744386/74296257-2ba06d80-4da8-11ea-945f-758d57221ec6.png)

I can also see the offset transaction created from it (in this case a cc/ customer credit)
![image](https://user-images.githubusercontent.com/29744386/74296309-45da4b80-4da8-11ea-9633-1e43c1a8e6ce.png)

If I open the transaction and close it again, the total is set to 0: 
![image](https://user-images.githubusercontent.com/29744386/74296330-55f22b00-4da8-11ea-88d3-9c0c542f9dbb.png)

### To reproduce

1. Create a cash transaction through the cash register in mobile. Sync to desktop
2. On desktop, log into the mobile store. Check the cash register. Make sure the total is correct. 
3. Find the customer credit or customer invoice created from the cash transaction you made on mobile. Open it and close it. 
4. Now check the total of that customer credit/customer invoice. 

### Expected behaviour

The values should remain the same 

### Proposed Solution

Leave if you don't know how to fix/implement. Edit this issue description and explain here if you know the best path of implementing the fix within the codebase.

### Version and device info

- App version: 4.0.0 RC4
- Tablet model:
- OS version:

**Comment by wlthomson** [comment]
Initial thoughts are that desktop is "recalculating" the total on save. Perhaps something related to missing `trans_line` fields. I think good place to start would be to trace the OK logic on desktop and see where the total is getting reassigned?

**Comment by DhanyaHerath** [comment]
Yeah, so I'm not sure if `[transact]subtotal` gets assigned in mobile, but [transact]total does get recalculated here: 
https://github.com/sussol/msupply/blob/a1d9bd6b89c4fa7c35acf5fc00c6a389dce3b1d0/vc_methods/trans_calc_totals.txt#L71

**Comment by wlthomson** [comment]
Update:

Investigated w/ @DhanyaHerath. Culprit is the following line in `trans_calc_totals`:

```
[transact]subtotal:=Sum([trans_line]price_extension)+[transact]service_price
```

Suggested fix is to update mobile code to sync `service_price` for cash transaction offset transactions.

**Comment by joshxg** [comment]
Closed by #2256
